akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -539,10 +542,13 @@ private[client] object NewHostConnectionPool {
           responseIn.pull()
 
           slot.debug("Establishing connection")
+          import Attributes.CancellationBehavior
+          import CancellationBehavior._
           val connection =
             Source.fromGraph(requestOut.source)
               .viaMat(connectionFlow)(Keep.right)
               .to(responseIn.sink)
+              .addAttributes(Attributes(CancellationBehavior(AfterDelay(1.second, FailStage))))
```

Comment:
This delays cancellation in the connection flow between all stages. We should add this attribute wherever we run or build streams that contain BidiStages that might exhibit the problematic behavior.